### PR TITLE
[TF:TRT] Enable support for TensorRT 8.2

### PR DESCRIPTION
**Pull request by christopherbate**
Add support for TensorRT 8.2.

Add plugin stubs for TensorRT 8.2.
Modify tests to hold an IBuilder to prevent TensorRT 8.2 from releasing builder resources when running parameterized tests.

### CODE REVIEW

**Comment by christopherbate** [comment]
Request review from @bixia1

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc [hunk 1]
```diff
@@ -143,6 +145,25 @@ void ValidateWeights(const TRT_ShapedWeights& weights,
   }
 }
 
+// TRT >= 8.2 optimized memory management in the builder. When all builders
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc:148]: s/optimized/optimizes/

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc [hunk 2]
```diff
@@ -143,6 +145,25 @@ void ValidateWeights(const TRT_ShapedWeights& weights,
...
+// are destroyed, it unloads many resources. This test fixtures will create and
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc:149]: s/fixtures/fixture/

**Comment by bixia1** [comment]
I edited the PR description with details on what the PR does. Please check.

**Review by bixia1 — APPROVED** [review]

**Comment by bixia1** [comment]
within google, we use TensorRT 7, and here is the error:
```
In file included from third_party/tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc:16:
In file included from ./third_party/tensorflow/compiler/tf2tensorrt/convert/convert_nodes.h:19:
In file included from third_party/stl/cxx17/set:16:
In file included from third_party/crosstool/v18/stable/toolchain/bin/../include/c++/v1/set:439:
In file included from third_party/crosstool/v18/stable/toolchain/bin/../include/c++/v1/__node_handle:63:
In file included from third_party/stl/cxx17/memory:4:
In file included from third_party/crosstool/v18/stable/toolchain/bin/../include/c++/v1/memory:682:
In file included from third_party/crosstool/v18/stable/toolchain/bin/../include/c++/v1/__memory/shared_ptr.h:25:
third_party/crosstool/v18/stable/toolchain/bin/../include/c++/v1/__memory/unique_ptr.h:60:12: error: calling a protected destructor of class 'nvinfer1::IBuilder'
    delete __ptr;
           ^
third_party/crosstool/v18/stable/toolchain/bin/../include/c++/v1/__memory/unique_ptr.h:282:7: note: in instantiation of member function 'std::default_delete<nvinfer1::IBuilder>::operator()' requested here
      __ptr_.second()(__tmp);
      ^
third_party/tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc:155:61: note: in instantiation of member function 'std::unique_ptr<nvinfer1::IBuilder>::~unique_ptr' requested here
  static std::unique_ptr<nvinfer1::IBuilder> hold_builder = nullptr;
                                                            ^
blaze-out/k8-cuda11-opt/bin/third_party/tensorrt/_virtual_includes/nv_infer_static_cuda_11_0/third_party/tensorrt/NvInfer.h:7097:13: note: declared protected here
    virtual ~IBuilder()
            ^
```

**Comment by christopherbate** [comment]
I pushed a change that should fix the issue.

**Review by bixia1 — APPROVED** [review]

**Review by bixia1 — APPROVED** [review]

**Pull request merged**